File: 1 - Arrays & Hashing/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complementMap = {}

        # using enumerator: 53 & 9
        # for i in range(len(nums)):
        #     currComplement = target - nums[i]
        #     if nums[i] in complementMap:
        #         return [i, complementMap[nums[i]]] 
        #     else:
        #         complementMap[currComplement] = i

        # using enumerator: 62 & 14
        for i,n in enumerate(nums):
            currComplement = target - n
            if n in complementMap:
                return [i, complementMap[n]] 
            else:
                complementMap[currComplement] = i

File: 1 - Arrays & Hashing/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> complementMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;
            if (complementMap.containsKey(num)) {
                return new int[] {i, complementMap.get(num)};
            }
            complementMap.put(diff, i);
            System.out.println(complementMap);
        }
        return new int[] {};
    }
}